src/v/cluster/members_table.cc

```diff
@@ -190,7 +193,9 @@ members_table::apply(model::offset version, maintenance_mode_cmd cmd) {
     const auto other = std::find_if(
       _brokers.cbegin(), _brokers.cend(), [](const auto& b) {
           return b.second->get_maintenance_state()
-                 == model::maintenance_state::active;
+                   == model::maintenance_state::active
+                 && b.second->get_membership_state()
+                      != model::membership_state::removed;
```

Comment:
a couple questions about this mechanism.

1. this is effectively "hiding" any node that has been removed so that a removed node does not affect the one-at-a-time decision?
2. does this "hiding" also need to be applied when reporting nodes that are maintenance mode?
3. since this is before the large refactor on dev, are we still dealing with a single source of truth here? that is once a broker eventually is removed from the set of brokers in the members table then it's also implicitly taking that broker out of maintenance mode?

Reply:
>this is effectively "hiding" any node that has been removed so that a removed node does not affect the one-at-a-time decision?

Right, removed nodes shouldn't be taken into account then.

>does this "hiding" also need to be applied when reporting nodes that are maintenance mode?

I think this is already done. At least, looking at the admin endpoint, we call `members_table::all_brokers()` (in v22.3 and below) which filters out nodes in the `removed` state.

>that is once a broker eventually is removed from the set of brokers in the members table then it's also implicitly taking that broker out of maintenance mode?

Right, by ignoring `removed` nodes (or by straight out removing them from the nodes list in `dev`), such nodes no longer count towards our checks towards nodes in maintenance.